packages/@aws-cdk/core/lib/arn.ts

```diff
@@ -76,11 +76,20 @@ export class Arn {
    * the 'scope' is attached to. If all ARN pieces are supplied, the supplied scope
    * can be 'undefined'.
    */
-  public static format(components: ArnComponents, stack: Stack): string {
-    const partition = components.partition !== undefined ? components.partition : stack.partition;
-    const region = components.region !== undefined ? components.region : stack.region;
-    const account = components.account !== undefined ? components.account : stack.account;
-    const sep = components.sep !== undefined ? components.sep : '/';
+  public static format(components: ArnComponents, stack?: Stack): string {
+    const requiredComponents: Array<keyof ArnComponents> = ['partition', 'region', 'account'];
+    if (stack === undefined) {
+      requiredComponents.forEach(c => {
```

Comment:
I prefer a normal "for" loop over `forEach`

Reply:
What are your feelings on reverse do-while loops? 😉